packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -126,9 +126,10 @@ const FocusScope = React.forwardRef((props, forwardedRef) => {
         // See: https://github.com/facebook/react/issues/17894
         setTimeout(() => {
           const unmountEvent = new Event(AUTOFOCUS_ON_UNMOUNT, EVENT_OPTIONS);
+          container.addEventListener(AUTOFOCUS_ON_UNMOUNT, onUnmountAutoFocus);
```

Comment:
@benoitgrelard I've moved this here where it's relevant because I accidentally wrapped it in the `hasFocusedCandidate` conditional before 🙈